corehq/form_processor/submission_post.py

```diff
@@ -445,6 +448,34 @@ def do_post_save_actions(case_db, xforms, case_stock_result):
             })
             raise PostSaveError
 
+    @staticmethod
+    def send_to_elasticsearch(instance):
+        if instance.metadata.deviceID != FORMPLAYER_DEVICE_ID:
+            return
+
+        domain_obj = Domain.get_by_name(instance.domain)
+        if not domain_obj or not domain_obj.web_apps_sync_case_search:
+            return
+
+        from casexml.apps.case.xform import extract_case_blocks
+        from corehq.elastic import send_to_elasticsearch
+        from corehq.pillows.case_search import transform_case_for_elasticsearch
+        case_ids = [info.get('@case_id') for info in extract_case_blocks(instance)]
+        cases = CommCareCase.objects.get_cases(case_ids)
+        for instance_case in cases:
```

Comment:
This could be done in bulk:
```python
actions = [BulkActionItem.index(transform_case_for_elasticsearch(case.to_json()) for case in cases]
try:
    _, errors = ElasticCaseSearch().bulk(actions, refresh=True, raise_on_error=False, raise_on_exception=False)  # I'm not sure if `refresh` is needed
except Exception as e:
    log
if errors:
    log
```

Reply:
We might wanna run that decision by SaaS, as the ES docs say it's resource intensive.  I get the impression it's an action on the whole index, so the slowness wouldn't just be the individual form submission, but the entire ES cluster.  If moving it from 5s to 1 is sufficient, that'd be a huge difference in number of refreshes.
Also, fwiw, we save users to ES synchronously for the same reasons, and we don't do a refresh there:
https://github.com/dimagi/commcare-hq/blob/ec31c5ed116d989b8904582014e463cbf57773b6/corehq/apps/users/signals.py#L30-L34
(though I think the UI might've changed since then, so we might not as readily notice a 5s delay anymore